lib/reverse_words.rb

```diff
@@ -1,6 +1,43 @@
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2), where n represents the length of the string. The outer loop takes O(n), and worst case scenario of the inner loops take O(n) and O(n/2), respectively. An O(n) loop inside of an O(n) loop results in O(n^2).
+# Space complexity: O(C), because the storage of variables i, j, c, b, temp doesn't change based on string length.
+
+# partial reverse string helper method
+def reverse_partial_word(string, beg_idx, end_idx)
+  while beg_idx < end_idx
+    temp = string[beg_idx]
+    string[beg_idx] = string[end_idx]
+    string[end_idx] = temp
+    beg_idx += 1
+    end_idx -= 1
+  end
+  return string
```

Comment:
I'd recommend renaming this variable. It's usually a bad idea to conflate a variable with a class.